tests/test_update_backups_schedule.py

```diff
@@ -0,0 +1,141 @@
+from unittest import mock
+
+import pytest
+from kubernetes_asyncio.client import CoreV1Api, CustomObjectsApi
+from kubernetes_asyncio.client.api.batch_v1_api import BatchV1Api
+
+from crate.operator.constants import API_GROUP, RESOURCE_CRATEDB
+from crate.operator.cratedb import connection_factory
+from crate.operator.webhooks import WebhookEvent, WebhookStatus
+
+from .utils import (
+    DEFAULT_TIMEOUT,
+    assert_wait_for,
+    create_test_sys_jobs_table,
+    is_cluster_healthy,
+    is_cronjob_schedule_matching,
+    start_cluster,
+    was_notification_sent,
+)
+
+
+@pytest.mark.k8s
+@pytest.mark.asyncio
+@mock.patch("crate.operator.webhooks.webhook_client._send")
+async def test_update_backups_schedule(
+    mock_send_notification,
+    faker,
+    namespace,
+    kopf_runner,
+    api_client,
+):
+
+    coapi = CustomObjectsApi(api_client)
+    core = CoreV1Api(api_client)
+    batch = BatchV1Api(api_client)
+    name = faker.domain_word()
+
+    backups_spec = {
+        "aws": {
+            "accessKeyId": {
+                "secretKeyRef": {
+                    "key": faker.domain_word(),
+                    "name": faker.domain_word(),
+                },
+            },
+            "basePath": faker.uri_path() + "/",
+            "cron": "1 2 3 4 5",
+            "region": {
+                "secretKeyRef": {
+                    "key": faker.domain_word(),
+                    "name": faker.domain_word(),
+                },
+            },
+            "bucket": {
+                "secretKeyRef": {
+                    "key": faker.domain_word(),
+                    "name": faker.domain_word(),
+                },
+            },
+            "secretAccessKey": {
+                "secretKeyRef": {
+                    "key": faker.domain_word(),
+                    "name": faker.domain_word(),
+                },
+            },
+            "endpointUrl": {
+                "secretKeyRef": {
+                    "key": faker.domain_word(),
+                    "name": faker.domain_word(),
+                    "optional": True,
+                },
+            },
+        },
+    }
+
+    host, password = await start_cluster(
+        name, namespace, core, coapi, 1, backups_spec=backups_spec
+    )
+    conn_factory = connection_factory(host, password)
+
+    await assert_wait_for(
+        True,
+        is_cluster_healthy,
+        conn_factory,
+        1,
+        err_msg="Cluster wasn't healthy",
+        timeout=DEFAULT_TIMEOUT,
+    )
+
+    await create_test_sys_jobs_table(conn_factory)
```

Comment:
You don't need the sys_jobs table, unless testing delays while backups are running.

Reply:
It was failing without it before, but it was due to beforeUpdate handler, which is not triggered anymore